Journalator/Source/Monitor/InvestONator.lua

```
@@ -0,0 +1,81 @@
+  
+  for portfolioId, portfolio in pairs(portfolios) do
+    if portfolio.items[itemID] then
+      -- Record the purchase
+      if Journalator.InvestONator.RecordPurchase(portfolioId, itemID, totalAmount, price, quantity) then
+        -- Re-fetch updated item data after mutation to get fresh remaining amount
+        local updatedPortfolio = Journalator.InvestONator.GetPortfolio and Journalator.InvestONator.GetPortfolio(portfolioId) or portfolio
+        local updatedItem = updatedPortfolio and updatedPortfolio.items and updatedPortfolio.items[itemID]
+        local remainingCopper = updatedItem and updatedItem.remainingAmount or 0
+        local remaining = Journalator.InvestONator.FormatGold(remainingCopper)
+
+        Journalator.Utilities.Message(string.format(
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'Journalator'

(W113)